packages/cordova-plugin-aerogear-metrics/www/metrics.js

```diff
@@ -0,0 +1,42 @@
+
+/**
+ * Get device metrics, to be called after deviceReady event
+ * 
+ * It uses cordova-plugin-device plugin.
+ * @example
+ * const deviceMetrics = await aerogear.getDeviceMetrics();
+ * @returns {Promise<DeviceMetrics>} The device metrics
+ */
+exports.getDeviceMetrics = function () {
```

Comment:
Any reason why we have logic in the plugin? 
Could this code be in the Cordova TypeScript wrapper.
`
platform: device.platform,
platformVersion: device.version,
device: device.model
`

Reply:
Actually it's ok to have it here.

Reply:
That's a very good idea I didn't think of, we have already a specific Metrics implementation `CordovaAppMetrics` and `CordovaDeviceMetrics` and this logic belongs there. The JS interface is therefore not necessary.

Reply:
+1 to make this just a bag of dependencies and have core use them directly.

Reply:
+1 on just having one place define the implementation of `CordovaAppMetrics` and `CordovaDeviceMetrics`, but I am thinking if it's better to define them inside the plugin. There are 2 reasons:

1. They are cordova-specific implementation and they rely on the cordova plugins to work. From that point of, it makes sense for the code to live as part of the plugin.
2. Now the plugin is completely self-contained and it can be used on its own (although this one probably doesn't really matter as we don't have a use case to use the plugin by itself.)

That said, I think there might be some technical challenges to implement it this way - like how to invoke the platform specific implementation if they are defined inside the plugin. Not sure if it's possible, but if we can do it, I think it will make the code cleaner.

Reply:
ok, I think that makes sense.